src/View/Requirements_Backend.php

```diff
@@ -494,9 +494,47 @@ protected function getAllJavascript()
     public function customScript($script, $uniquenessID = null)
     {
         if ($uniquenessID) {
-            $this->customScript[$uniquenessID] = $script;
+            $this->customScript[$uniquenessID]['content'] = $script;
         } else {
-            $this->customScript[] = $script;
+            $this->customScript[]['content'] = $script;
+        }
+    }
+
+    /**
+     * Register the given Javascript code into the list of requirements with optional tag
+     * attributes.
+     *
+     * @param string     $script       The script content as a string (without enclosing `<script>` tag)
+     * @param array $options List of options. Available options include:
+     * - 'type' : Specifies the type of script
+     * - 'crossorigin' : Cross-origin policy for the resource
+     * @param string|int $uniquenessID A unique ID that ensures a piece of code is only added once
+     */
+    public function customScriptWithAttributes($script, $options = [], $uniquenessID = null)
+    {
+        // Get type
+        $type = null;
```

Comment:
Why restrict to 'type' and 'crossorigin'? Just let users add in whatever attributes they want?

You'll also what to strtolower() all attribute keys

Reply:
Sure thing I can update this, though my original reasoning for this is so it follows Requirements::javascript handling of attributes. This only allows attributes that work in an inline script tag. 

Below is how Requirements::javascript handle tags. I could make another issue to have both customScriptWithAttributes and javascript follow the same handling principals for attributes?
https://github.com/silverstripe/silverstripe-framework/blob/809f9e7ae06e4e3999ef2670aa322aacdddb7def/src/View/Requirements_Backend.php#L395-L404

Reply:
I'm not sure how much sense it makes to get consistency between the two, off the top of my head I don't know possible script attributes are relevant for inline scripts vs scripts with a src attribute, so I'd say don't worry about getting to two consistent.

If nothing else restricting possible attribute might be good to reduce any weird security vulnerabilities, so perhaps ignore my 'allow everything' recommendation and leave this as is for now with it restricted to 'type' and 'crossorigin' - we can add more allowed keys later on

Reply:
Out of scope for this PR, but an "unsafe" option/setting may be a good solution to allowing vulnerable tag attributes. I am currently using Vue with the SilverStripe framework, there are many tag attributes specific to Vue as well as props. 

So far I haven't had to worry about using them with a script tag though, but having the option may provide ease of use for integrating projects or custom solutions that make use of custom attributes.